kodilla-stream/src/test/java/com/kodilla/stream/portfolio/BoardTestSuite.java

```diff
@@ -135,4 +140,23 @@ public void testAddTaskListFindLongTasks() {
         //Then
         Assert.assertEquals(2, longTasks);
     }
+
+    @Test
+    public void testAddTaskListAverageWorkingOnTask() {
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        List<TaskList> inProgressTasks = new ArrayList<>();
+        inProgressTasks.add(new TaskList("In progress"));
+        double averageWorkingOnTask = project.getTaskLists().stream()
+                .filter(inProgressTasks::contains)
+                .flatMap(n -> n.getTasks().stream())
+                .mapToInt(n -> Period.between(n.getCreated(), LocalDate.now()).getDays())
+                .average()
+                .getAsDouble();
```

Comment:
Użycie `getAsDouble()` tutaj jest niebezpieczne - w przypadku, gdyby było 0 zadań na liście, dostaniesz wyjątek.
Lepiej użyć `.orElse(<tu wartość domyślna w przypadku braku elementów>)`.
Np. możesz wtedy zwrócić 0.0, nie jest to ścisłe z matematycznego punktu widzenia ale ma sens - nie ma zadań no to średni czas wynosi 0.

Reply:
zmienionw